sdk/search/azure-search/src/main/java/com/azure/search/models/EntityCategory.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Defines values for EntityCategory.
+ */
+public enum EntityCategory {
```

Comment:
We may want to adjust the swagger to make this and some of the other skill based fields string based (with a list of allowed values like in DataSourceType) instead of an enum.

Reply:
@careyjmac I recall @arv100kri scrubbed the Swagger for this already. The guidance is to use enums when the set of options is not-open ended and changes infrequently, otherwise use extensible enums.

Reply:
I think all of these skill related enums should be extensible, then.  Cognitive services is constantly adding new features that we want to be able to support with our existing skills, and in those cases it usually just involved supporting a new type in a pre existing property, which should therefore be backwards compatible.

Reply:
@careyjmac Please ensure there are work items for your team to update the Swagger accordingly. This should happen soon, since work on other Track 2 SDKs is commencing.